File: 747-MinCostClimbingStairs/747-MinCostClimbingStairs.py
# Last updated: 8/12/2026, 11:28:47 AM
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:

        l=len(cost)

        dp=[0]*(l+1)

        dp[0]=cost[0]
        dp[1]=cost[1]

        for i in range(2,len(cost)):
            dp[i]=cost[i]+min(dp[i-1],dp[i-2])

        return min(dp[l-1],dp[l-2])
        

File: 747-MinCostClimbingStairs/747-MinCostClimbingStairs.java
// Last updated: 8/12/2026, 11:28:42 AM
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        if(cost.length==1){
            return cost[0];
        }

        int [] dp=new int[cost.length];
        int n=cost.length;
        
        dp[0]=cost[0];
        dp[1]=cost[1];
             

        for(int i=2;i<cost.length;i++){
            dp[i]=cost[i]+Math.min(dp[i-1],dp[i-2]);

        }

        return Math.min(dp[n-1],dp[n-2]);
    }
}